### PR TITLE
ipc.susu_2327

### PR DESCRIPTION
**Pull request by Dirwul**
accepted

### CODE REVIEW

File: ipc/susu/task_2327/solve.cpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	int nComponents, nTypes, maxSum;
+	scanf("%d %d", &nTypes, &nComponents);
+	std::vector<std::vector<Component>> types(nTypes);
+	for (int i = 0; i < nComponents; i++) {
+		int cost, rate, type;
+		scanf("%d %d %d", &cost, &rate, &type);
+		types[type - 1].push_back(Component{ cost, rate, i + 1});
+	}
+	scanf("%d", &maxSum);
+	std::vector<std::vector<int>> d(nTypes + 1, std::vector<int>(maxSum + 1, -1));
```
Comment (terradrenor) [inline ipc/susu/task_2327/solve.cpp]: rename d

File: ipc/susu/task_2327/solve.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		d[0][i] = 0;
+	}
+	std::vector<std::vector<int>> usedComponent(nTypes + 1, std::vector<int>(maxSum + 1));
+	for (int iType = 0; iType < nTypes; iType++) {
+		for (int sum = 0; sum < maxSum; sum++) {
+			for (int iComponent = 0; iComponent < (int)types[iType].size(); iComponent++) {
+				if (iType == 0 || d[iType][sum] != 0) {
+					int nextSum = sum + types[iType][iComponent].cost;
+					if (nextSum <= maxSum && 
+						d[iType + 1][nextSum] < d[iType][sum] + types[iType][iComponent].rate) {
```
Comment (terradrenor) [inline ipc/susu/task_2327/solve.cpp]: fix code-style

File: ipc/susu/task_2327/solve.cpp (hunk 1 above)
Comment (Dirwul) [reply ipc/susu/task_2327/solve.cpp]: done

File: ipc/susu/task_2327/solve.cpp (hunk 2 above)
Comment (Dirwul) [reply ipc/susu/task_2327/solve.cpp]: done

**Review by terradrenor — APPROVED** [review]

**Pull request merged**